### DISCUSSION TITLE
`JAVA_HOME` env variable issue.

### DISCUSSION THREAD

**Discussion by sureshg** [community/community#160592] [Actions]
Java versions 8, 11, 17 & 21 are installed by default on Github action runners, which is awesome and also it exposes `JAVA_HOME` env variable for each version. But i don't understand the reasoning behind appending the CPU arch along with the env name (https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#java). This will cause some headache when  writing actions as we have to form the proper env variable based on CPU Arch (M1 mac, ARM64). Moreover on ARM64, it uses lowercase, so can't directly use `RUNNER_ARCH` env varaible.

Version | Environment Variable
-- | --
8.0.392+8 | JAVA_HOME_8_X64
11.0.21+9 (default) | JAVA_HOME_11_X64
17.0.9+9 | JAVA_HOME_17_X64
21.0.1+12 | JAVA_HOME_21_X64

* macos-14

Version | Environment Variable
-- | --
11.0.21+9 | JAVA_HOME_11_arm64
17.0.9+9 | JAVA_HOME_17_arm64
21.0.1+12.0 (default) | JAVA_HOME_21_arm64

So is it possible to add new `JAVA_HOME` env variables without the CPU arch appended to it. This will make [writing actions using java](https://github.com/oracle-actions/setup-java/pull/71/files) a bit more easier.

**Comment by erik-bershel** [comment]
Hey @sureshg!
Valid question and concern. Unfortunately the problem is that too many people rely on an old variable that contains an architecture reference. The impact of such changes will be enormous. Will think about adding duplicate variable without architect reference.

**Reply by gdams** [reply]
please do add this support.This would be less of a problem if we could rely on ${{ runner.arch }}

E.g `JAVA_HOME_21_${{ runner.arch }}`

The problem is that while this maps on `X64` for some reason the runner images use the lowercase `arm64` rather than following the convention with `ARM64`. Perhaps this is the issue that needs fixing instead?

**Reply by eregon** [reply]
BTW on [ubuntu-24.04-arm](https://github.com/actions/partner-runner-images/blob/main/images/arm-ubuntu-24-image.md) and [ubuntu-22.04-arm](https://github.com/actions/partner-runner-images/blob/main/images/arm-ubuntu-22-image.md) the variables are `JAVA_HOME_$VERSION_X64` e.g. https://github.com/eregon/actions-shell/actions/runs/13798182243/job/38594765318#step:4:5

:+1: to have aliases without the architecture in it, the architecture is redundant and used inconsistently.

**Comment by Shchvova** [comment]
Omg, this was driving me mad :( Thanks for hits. Fixed with 
```
JAVA_HOME="${JAVA_HOME_17_X64:-$JAVA_HOME_17_arm64}" ./build.sh
```
